### PR TITLE
stm32mp1: fix PSCI_CPU_OFF and bring PSCI_SYSTEM_RESET

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

**Comment by etienne-lms** [comment]
forced update: wrong version initially posted.
Damn, Travis has already failed...
```
ba3899ce plat-stm32mp1: PSCI_SYSTEM_RESET support
CHECK: usleep_range is preferred over udelay; see Documentation/timers/timers-howto.rst
#30: FILE: core/arch/arm/plat-stm32mp1/pm/psci.c:215:
+	udelay(100);
```

**Comment by jforissier** [comment]
> Damn, Travis has already failed...
> 
> ```
> ba3899ce plat-stm32mp1: PSCI_SYSTEM_RESET support
> CHECK: usleep_range is preferred over udelay; see Documentation/timers/timers-howto.rst
> #30: FILE: core/arch/arm/plat-stm32mp1/pm/psci.c:215:
> +	udelay(100);
> ```

Look like we should add `--ignore USLEEP_RANGE` to `scripts/checkpatch_inc.sh`. This is clearly applicable to Linux only.

**Comment by etienne-lms** [comment]
review tags applied. 

> Look like we should add --ignore USLEEP_RANGE to scripts/checkpatch_inc.sh. This is clearly applicable to Linux only.

Created https://github.com/OP-TEE/optee_os/pull/3812.

File: core/arch/arm/plat-stm32mp1/pm/psci.c [hunk 1]
```diff
@@ -204,16 +204,31 @@ int psci_cpu_off(void)
...
+	case PSCI_SYSTEM_RESET:
 	case PSCI_VERSION:
-#if CFG_TEE_CORE_NB_CORE > 1
-	case PSCI_CPU_ON:
-#endif
 		return PSCI_RET_SUCCESS;
+	case PSCI_CPU_ON:
+	case PSCI_CPU_OFF:
+		if (CFG_TEE_CORE_NB_CORE > 1)
+			PSCI_RET_SUCCESS;
```
Comment (jforissier) [inline core/arch/arm/plat-stm32mp1/pm/psci.c:230]: `return` :open_mouth:

**Comment by jforissier** [comment]
With the missing return added, and rebased onto master, CI should be fine.

**Comment by etienne-lms** [comment]
oh! sorry.

**Comment by etienne-lms** [comment]
rebased. missing return fixed. review tags still there.

**Comment by jforissier** [comment]
All good, thanks.

**Pull request merged**